src/Altinn.Common/Altinn.Common.AccessToken/Altinn.Common.AccessTokenClient/Services/AccessTokenGenerator.cs

```diff
@@ -79,5 +79,55 @@ public string GenerateAccessToken(string issuer, string app)
 
             return null;
         }
+
+        /// <summary>
+        /// Generates a access token for apps/functions needing to access platform components.
+        /// </summary>
+        /// <param name="issuer">Can be a app or platform component</param>
+        /// <param name="app">The application creating token (app or component)</param>
+        /// <param name="keyVaultUri">Uri to keyvault</param>
+        /// <param name="secretId">Id to the certificate in keyvault</param>
+        /// <returns>Accesstoken</returns>
+        public string GenerateAccessToken(string issuer, string app, string keyVaultUri, string secretId)
```

Comment:
a lot of duplication here. How about making kvUri and secretId optional params in the previous method and have logic to handle retrieving SigningCredentials with different methods to avoid this much duplication?

Reply:
Changed